### ISSUE TITLE
[Question] Ciena Config base setup

### ISSUE THREAD

**Issue by georgesnow** [ktbyers/netmiko#1422]
@ktbyers I started working on the Ciena config pieces. I was looking for input on the implementation. There are few pieces of Ciena that seem different than most other vendors.  From my testing it seems netmiko expects to enter config mode to run config commands from file. In the Ciena device case this presenet some hurdles to make it work reliably. And also I don't want to make hacky implentation. 

if you don't mind sharing your expectations before I re-start my effort.
Thanks.

Caveats:
-When connecting the immediate prompt is essentially "config mode". i.e.. you can start issuing commands to make changes that are "live" and active (however not saved on reboot)
-After issuing a command that changes the config. The prompt adds a * before > in the prompt

> Before command: 
> device>
> After:
> device*>

-When you issue config save the * goes away returns back to just >

**Comment by ktbyers** [comment]
@georgesnow This is a different driver than the Ciena saos driver?

**Comment by georgesnow** [comment]
nope, this is the ciena SAOS driver.

**Comment by ktbyers** [comment]
I thought we had this all worked out including `config` operations? So you are saying something is still broken.

`config_mode` on the Ciena Saos does nothing (as the driver knows there is no configuration mode).

The general process should be to use the `send_config_set` method and send a list of configuration commands.

Have you tried to do that?

Do you get an exception/errors when doing that.


Regards, Kirk

**Comment by georgesnow** [comment]
First thing. I will re-run the tests for the` send_config_set` cause I could be mistaken. When I had  initially tested the send_config_set methods (previous to all the work we had done). They didn't seem work.

But I think the confusion is I thought we were only addressing the `config` operations that were **specific to** the SCP/File Transfer operations. 

At the time when we went to merge the code I was already working on SCP trying fixing the formatting  **as well as** the config_mode, send_config_set operations. And when you started to fixing the SCP formatting I bailed _**on all those changes including the config operations**_ in effort to not duplicate effort and create other issues. And I had plan to go back and fix the _other config operations_ later.
